db/migrations/20200603124229_create_time_ilk_snapshot_query.sql

```diff
@@ -0,0 +1,66 @@
+-- +goose Up
+CREATE TYPE api.time_ilk_snapshot AS
+(
+    bucket_start TIMESTAMP,
+    bucket_end TIMESTAMP,
+    bucket_interval INTERVAL,
+    ilk_identifier TEXT,
+    block_number BIGINT,
+    rate NUMERIC,
+    art NUMERIC,
+    spot NUMERIC,
+    line NUMERIC,
+    dust NUMERIC,
+    chop NUMERIC,
+    lump NUMERIC,
+    flip TEXT,
+    rho NUMERIC,
+    duty NUMERIC,
+    pip TEXT,
+    mat NUMERIC,
+    created TIMESTAMP,
+    updated TIMESTAMP
+);
+
+CREATE FUNCTION api.time_ilk_snapshots(ilk_identifier TEXT, bucket_start TIMESTAMP, bucket_end TIMESTAMP, bucket_interval INTERVAL)
+    RETURNS SETOF api.time_ilk_snapshot AS
+$$
+WITH buckets AS (SELECT generate_series(bucket_start, bucket_end - bucket_interval, bucket_interval) AS bucket_start)
```

Comment:
Is the term `bucket_start` being overloaded here? Maybe we could refer to one as `time_ilk_snapshots.bucket_start` to clarify which value is coming from where?

Reply:
You are correct - bucket_start was overloaded, and it's not really accurate for the function argument case. I changed the arguments to be range_start and range_end as it is not the bucket start and end.